src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java

```diff
@@ -161,25 +162,42 @@ public static void scan(Run build, PrintStream buildLog) {
 
             if (!downstreamFailureCauses.isEmpty()) {
                 buildLog.println("[BFA] Found downstream Failure causes ...");
-                for (FailureCauseDisplayData displayData : downstreamFailureCauses) {
-                  FailureCauseDisplayData.Links links = displayData.getLinks();
-                  buildLog.println("[BFA] See " + links.getProjectDisplayName() + links.getBuildDisplayName());
-                  List<FoundFailureCause> failureCauses = displayData.getFoundFailureCauses();
-                  for (FoundFailureCause foundCause : failureCauses) {
-                    String foundString = "[BFA] " + foundCause.getName();
-                    if (foundCause.getCategories() != null) {
-                        foundString += " from category " + foundCause.getCategories().get(0);
-                    }
-                    buildLog.println(foundString);
-                    }
-                }
+                printDownstream(buildLog, downstreamFailureCauses);
             }
+
             StatisticsLogger.getInstance().log(build, foundCauseListToLog);
         } catch (Exception e) {
             logger.log(Level.SEVERE, "Could not scan build " + build, e);
         }
     }
 
+
+    /**
+     *
+     * Adds all causes from downstream builds in recursion
+     *
+     * @param buildLog log to write information to.
+     * @param downstreamFailureCauses the list of downstream failure causes.
+     */
+    private static void printDownstream(PrintStream buildLog, List<FailureCauseDisplayData> downstreamFailureCauses) {
+        if (!downstreamFailureCauses.isEmpty()) {
+            for (FailureCauseDisplayData displayData : downstreamFailureCauses) {
```

Comment:
Yes, you are right. I can skip check inside method. I don't want to skip check before calling method, because it's needed for nice print:
```
buildLog.println("[BFA] Found downstream Failure causes ...");
```